### PR TITLE
Reduce code complexity in DataBlock.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6028 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/DataBlock.js [hunk 1]
```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
...
-			>
-				<div className="googlesitekit-data-block__title-datapoint-wrapper">
-					<h3
-						className="
+			) }
+			tabIndex={ isButtonContext && ! gatheringData ? '0' : '-1' }
+			role={ handleStatSelection && role }
+			onClick={ handleClick }
+			onKeyDown={ handleKeyDown }
+			aria-disabled={ gatheringData }
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock.js:121]: This property value should remain the same as it was before:
```suggestion
			aria-disabled={ gatheringData || undefined }
```

File: assets/js/components/DataBlock.js [hunk 2]
```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
...
+	period = '',
+	selected = false,
+	source,
+	sparkline,
+	handleStatSelection,
+	invertChangeColor = false,
+	gatheringData = false,
+	gatheringDataNoticeStyle = NOTICE_STYLE.DEFAULT,
+} ) => {
+	const handleClick = () => {
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock.js:57]: Please, use `useCallback` for all callbacks to make it consistent with other components.

File: assets/js/components/DataBlock.js [hunk 3]
```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
...
-				onClick={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleClick
-				}
-				onKeyPress={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleKeyPress
+	const datapointFormatted = datapoint && numFmt( datapoint, datapointUnit );
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock.js:101]: Same here, I don't know for sure but explicitly checking whether the `datapoint` variable is `undefined` might have some sense that will be lost after your changes. Please, revert it as it was before.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. Left a few comments for you. Please, take a look.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. Looks good to me, but I am not sure about naming. The `DataBlockAddons` name is not clear to me. Let's probably move these components into `components/DataBlock` folder and split `DataBlockAddons` into multiple components: `Sparkline`, `Change` and `SourceLink`.

File: assets/js/components/DataBlock/Change.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Change component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock/Change.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2023 Google LLC
```

File: assets/js/components/DataBlock/Change.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		changeFormatted = sprintf( period, changeFormatted );
+	}
+
+	return (
+		<div
+			className={ classnames( 'googlesitekit-data-block__change', {
+				'googlesitekit-data-block__change--no-change': ! change,
+			} ) }
+		>
+			<Fragment>
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock/Change.js:65]: We don't need this fragment element. Let's get rid of it.

File: assets/js/components/DataBlock/index.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
+/**
+ * DataBlock component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock/index.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2023 Google LLC
```

File: assets/js/components/DataBlock/index.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+							<SourceLink
+								className="googlesitekit-data-block__source"
+								name={ source.name }
+								href={ source.link }
+								external={ source?.external }
+							/>
+						) }
+					</div>
+				</Fragment>
+			) }
```
Comment (eugene-manuilov) [inline assets/js/components/DataBlock/index.js:144]: Let's split it to remove a need for using the `Fragment` element:

```js
{ ! gatheringData && sparkline && (
    <Sparkline ... />
) }
{ ! gatheringData && (
    <div className="googlesitekit-data-block__change-source-wrapper">...</div>
) }
{ gatheringData && (
    <GatheringDataNotice style={ gatheringDataNoticeStyle } />
) }
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. Added the last comments from myself. Once they are addressed, we are good to go.

**Comment by sashadoes** [comment]
Thank you @eugene-manuilov 🙂

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @sashadoes. Looks good to me.

**Pull request merged**